File: 2018/problems/gatnamot/submissions/accepted/gardar_large.py
import sys
r = int(sys.stdin.readline())
atx = 0
aty = r
ans = 0
while atx <= r:
    while aty*aty + atx*atx > r*r:
        aty -= 1
    ans += aty
    atx += 1

ans *= 4
ans += 1

print(ans)

File: 2018/problems/gatnamot/submissions/accepted/gardar_large.cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;

int main(){
    ll r;
	cin >> r;

	ll atx = 0;
	ll aty = r;
	ll ans = 0;
	while(atx <= r){
		while(aty*aty + atx*atx > r*r){
			aty--;
		}
		ans += aty;
		atx++;
	}

	ans *= 4;
	ans++;

	cout << ans << endl;
}


